wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -674,6 +687,56 @@ public boolean isAnyInProgress() {
     return environmentFactories.keySet();
   }
 
+  void stop() {
+    Set<RuntimeIdentity> identities;
+    try {
+      identities = infrastructure.getIdentities();
+    } catch (UnsupportedOperationException e) {
+      LOG.warn("Not recoverable infrastructure: '{}'", infrastructure.getName());
+      return;
+    } catch (InfrastructureException e) {
+      LOG.error(
+          "An error occurred while attempting to get runtime identities for infrastructure '{}'. Reason: '{}'",
+          infrastructure.getName(),
+          e.getMessage());
+      return;
+    }
+
+    LOG.info(
+        "Infrastructure is tracking {} active runtimes that need to be stopped", identities.size());
+
+    if (identities.isEmpty()) {
+      return;
+    }
+
+    for (RuntimeIdentity identity : identities) {
+      try {
+        String workspaceId = identity.getWorkspaceId();
+        WorkspaceImpl workspace = workspaceDao.get(workspaceId);
+        try (Unlocker ignored = lockService.writeLock(workspaceId)) {
+          statuses.putIfAbsent(workspaceId, STARTING);
```

Comment:
why `STARTING` ?

Reply:
stopAsync expects this state I believe - https://github.com/eclipse-che/che-server/pull/177/files#diff-306aaf1c9fd22ee3217510b1b80ad0fff44f3b6d4afe2a95bb5b13e4426b5055R533

Reply:
it accepts both `RUNNING` and `STARTING`, other are ignored. So that status change seems to be redundant.

Reply:
well, but there could be a case when the status is blank / null, right? so my understanding that we need to put smth.